dotnet-trace should exit gracefully when runtime data stops

After the repro in #203 Andy observed this on the dotnet-trace output:

```
    Enabling the following providers
            Microsoft-Windows-DotNETRuntime:0x0000000000000001:4
            Microsoft-DotNETCore-SampleProfiler:0xFFFFFFFFFFFFFFFF:5
            Microsoft-Windows-DotNETRuntime:0x00000004C14FCCBD:4
    Press <Enter> or <Ctrl+C> to exit...
    Process     : C:\Users\anhans\work\coreclr\bin\tests\Windows_NT.x64.Debug\Tests\Core_Root\CoreRun.exe
    Output File : C:\Users\anhans\foo.netperf
            Recording trace 123.357 (KB)[ERROR] System.IO.EndOfStreamException: Unable to read beyond the end of the stream.
       at System.IO.BinaryReader.FillBuffer(Int32 numBytes)
       at System.IO.BinaryReader.ReadUInt64()
       at Microsoft.Diagnostics.Tools.RuntimeClient.EventPipeClient.SendCommand(Int32 processId, Byte[] buffer) in C:\Users\anhans\work\diagnostics\src\Microsoft.Diagnostics.Tools.RuntimeClient\Eventing\EventPipeClient.cs:line 38
       at Microsoft.Diagnostics.Tools.RuntimeClient.EventPipeClient.StopTracing(Int32 processId, UInt64 sessionId) in C:\Users\anhans\work\diagnostics\src\Microsoft.Diagnostics.Tools.RuntimeClient\Eventing\EventPipeClient.cs:line 169
       at Microsoft.Diagnostics.Tools.Trace.CollectCommandHandler.Collect(IConsole console, Int32 processId, FileInfo output, UInt32 buffersize, String providers, String profile, TraceFileFormat format) in C:\Users\anhans\work\diagnostics\src\Tools\dotnet-trace\CommandLine\Commands\CollectCommand.cs:line 108
```


We shouldn't be dumping exception stacks on the command line when the stream ends abruptly.

--- a/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
+++ b/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
@@ -86,6 +86,7 @@ namespace Microsoft.Diagnostics.Tools.Trace
                     providers: providerCollection);
 
                 var shouldExit = new ManualResetEvent(false);
+                var terminated = false;
 
                 ulong sessionId = 0;
                 using (Stream stream = EventPipeClient.CollectTracing(processId, configuration, out sessionId))
@@ -118,6 +119,8 @@ namespace Microsoft.Diagnostics.Tools.Trace
 
                                 Debug.WriteLine($"PACKET: {Convert.ToBase64String(buffer, 0, nBytesRead)} (bytes {nBytesRead})");
                             }
+                            terminated = true;
+                            shouldExit.Set();
                         }
                     });
                     collectingTask.Start();
@@ -132,7 +135,10 @@ namespace Microsoft.Diagnostics.Tools.Trace
                         while (!Console.KeyAvailable && !shouldExit.WaitOne(250)) { }
                     } while (!shouldExit.WaitOne(0) && Console.ReadKey(true).Key != ConsoleKey.Enter);
 
-                    EventPipeClient.StopTracing(processId, sessionId);
+                    if (!terminated)
+                    {
+                        EventPipeClient.StopTracing(processId, sessionId);
+                    }
                     collectingTask.Wait();
                 }
 

